src/tests/conformance/squelch.rs

```diff
@@ -0,0 +1,92 @@
+use tempfile::TempDir;
+use tokio::time::{sleep, timeout, Duration};
+
+use crate::{
+    protocol::{
+        codecs::binary::{BinaryMessage, Payload},
+        proto::{TmProposeSet, TmSquelch},
+    },
+    setup::node::build_stateful_builder,
+    tools::{rpc::wait_for_state, synth_node::SyntheticNode},
+};
+
+#[tokio::test]
+#[allow(non_snake_case)]
+async fn c009_TM_SQUELCH_cannot_squelch_neighbouring_ledger_proposals() {
+    // ZG-CONFORMANCE-009
+
+    // Time we shall wait for a TmProposeLedger message.
+    const WAIT_MSG_TIMEOUT: Duration = Duration::from_secs(10);
+
+    // Create a stateful node.
+    let target = TempDir::new().expect("unable to create TempDir");
+    let mut node = build_stateful_builder(target.path().to_path_buf())
+        .expect("unable to get stateful builder")
+        .start(true)
+        .await
+        .expect("unable to start stateful node");
+
+    // Wait for correct state and account data.
+    wait_for_state("proposing".into()).await;
+
+    // Connect synth node.
+    let mut synth_node = SyntheticNode::new(&Default::default()).await;
+    synth_node
+        .connect(node.addr())
+        .await
+        .expect("unable to connect");
+
+    // Get a validator public key.
+    let mut validator_pub_key: Vec<u8> = Vec::new();
+    timeout(WAIT_MSG_TIMEOUT, async {
+        loop {
+            if let (
+                _,
+                BinaryMessage {
+                    payload: Payload::TmProposeLedger(TmProposeSet { node_pub_key, .. }),
+                    ..
+                },
+            ) = synth_node.recv_message().await
+            {
+                validator_pub_key = node_pub_key;
+                break;
+            }
+        }
+    })
+    .await
+    .expect("TmProposeLedger not received in time");
+
+    // Squelch the validator public key belonging to our only neighbour.
+    let msg = Payload::TmSquelch(TmSquelch {
+        squelch: true,
+        validator_pub_key: validator_pub_key.clone(),
+        squelch_duration: Some(6 * 60), // Six minutes should be an ample time value.
```

Comment:
maybe we should make it a `const` at the beginning (like `WAIT_MSG_TIMEOUT`), for easier tweaking?

Reply:
Done